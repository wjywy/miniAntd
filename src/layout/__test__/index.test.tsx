import { render } from '@testing-library/react';
import React from 'react';
import { Content, Footer, Header, Layout, Sider } from '../Layout';
describe('Layout', () => {
  it('should render correctly', () => {
    const headerFooterStyle = {
      background: '#7dbcea',
      height: '64px',
      color: 'white',
      textAlign: 'center',
      lineHeight: '64px',
    };
    const contentStyle = {
      background: '#108ee9',
      height: '120px',
      color: 'white',
      textAlign: 'center',
      lineHeight: '120px',
    };
    const siderStyle = {
      background: '#3ba0e9',
      width: '120px',
      color: 'white',
      textAlign: 'center',
      lineHeight: '120px',
    };

    const { getByTestId } = render(
      <div data-testid="Layout">
        <Layout>
          <Header style={headerFooterStyle}>header</Header>
          <Content style={contentStyle}>content</Content>
          <Footer style={headerFooterStyle}>footer</Footer>
        </Layout>
        <br />
        <Layout>
          <Header style={headerFooterStyle}>header</Header>
          <Layout>
            <Sider style={siderStyle}>sider</Sider>
            <Content style={contentStyle}>content</Content>
          </Layout>
          <Footer style={headerFooterStyle}>footer</Footer>
        </Layout>
        <br />
        <Layout>
          <Header style={headerFooterStyle}>header</Header>
          <Layout>
            <Content style={contentStyle}>content</Content>
            <Sider style={siderStyle}>sider</Sider>
          </Layout>
          <Footer style={headerFooterStyle}>footer</Footer>
        </Layout>
        <br />
        <Layout>
          <Sider style={siderStyle}>sider</Sider>
          <Layout>
            <Header style={headerFooterStyle}>header</Header>
            <Content style={contentStyle}>content</Content>
            <Footer style={headerFooterStyle}>footer</Footer>
          </Layout>
        </Layout>
        <br />
        <span>实例</span>
        <Layout>
          <Header style={headerFooterStyle}>
            <ul className="layout-test">
              <li>nav1</li>
              <li>nav2</li>
              <li>nav3</li>
              <li>nav4</li>
              <li>nav5</li>
            </ul>
          </Header>
          <Content style={contentStyle}>content</Content>
          <Footer style={headerFooterStyle}>footer</Footer>
        </Layout>
      </div>,
    );
    expect(getByTestId('Layout')).toMatchSnapshot();
  });
});
