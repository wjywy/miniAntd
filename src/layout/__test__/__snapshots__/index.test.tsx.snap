// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<div
  data-testid="Layout"
>
  <div
    class="Layout-column"
  >
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      header
    </div>
    <div
      class="content"
      style="background: rgb(16, 142, 233); height: 120px; color: white; text-align: center; line-height: 120px;"
    >
      content
    </div>
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      footer
    </div>
  </div>
  <br />
  <div
    class="Layout-column"
  >
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      header
    </div>
    <div
      class="Layout-row"
    >
      <div
        style="background: rgb(59, 160, 233); width: 120px; color: white; text-align: center; line-height: 120px;"
      >
        sider
      </div>
      <div
        class="content"
        style="background: rgb(16, 142, 233); height: 120px; color: white; text-align: center; line-height: 120px;"
      >
        content
      </div>
    </div>
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      footer
    </div>
  </div>
  <br />
  <div
    class="Layout-column"
  >
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      header
    </div>
    <div
      class="Layout-row"
    >
      <div
        class="content"
        style="background: rgb(16, 142, 233); height: 120px; color: white; text-align: center; line-height: 120px;"
      >
        content
      </div>
      <div
        style="background: rgb(59, 160, 233); width: 120px; color: white; text-align: center; line-height: 120px;"
      >
        sider
      </div>
    </div>
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      footer
    </div>
  </div>
  <br />
  <div
    class="Layout-row"
  >
    <div
      style="background: rgb(59, 160, 233); width: 120px; color: white; text-align: center; line-height: 120px;"
    >
      sider
    </div>
    <div
      class="Layout-column"
    >
      <div
        style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
      >
        header
      </div>
      <div
        class="content"
        style="background: rgb(16, 142, 233); height: 120px; color: white; text-align: center; line-height: 120px;"
      >
        content
      </div>
      <div
        style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
      >
        footer
      </div>
    </div>
  </div>
  <br />
  <span>
    实例
  </span>
  <div
    class="Layout-column"
  >
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      <ul
        class="layout-test"
      >
        <li>
          nav1
        </li>
        <li>
          nav2
        </li>
        <li>
          nav3
        </li>
        <li>
          nav4
        </li>
        <li>
          nav5
        </li>
      </ul>
    </div>
    <div
      class="content"
      style="background: rgb(16, 142, 233); height: 120px; color: white; text-align: center; line-height: 120px;"
    >
      content
    </div>
    <div
      style="background: rgb(125, 188, 234); height: 64px; color: white; text-align: center; line-height: 64px;"
    >
      footer
    </div>
  </div>
</div>
`;
